qiskit/dagcircuit/dagcircuit.py

```diff
@@ -566,22 +566,30 @@ def _map_condition(wire_map, condition):
         else:
             # if there is a condition, map the condition bits to the
             # composed cregs based on the wire_map
-            creg = condition[0]
+            cond_creg = condition[0]
             cond_val = condition[1]
             new_cond_val = 0
             new_creg = None
             for bit in wire_map:
-                if bit in creg:
+                if bit in cond_creg:
+                    try:
+                        candidate_creg = next(creg
+                                              for creg in target_cregs
+                                              if wire_map[bit] in creg)
+                    except StopIteration:
+                        raise DAGCircuitError('Did not find creg containing '
+                                              'mapped clbit in conditional.')
+
                     if new_creg is None:
-                        new_creg = wire_map[bit].register
-                    elif new_creg != wire_map[bit].register:
+                        new_creg = candidate_creg
+                    elif new_creg != candidate_creg:
                         # Raise if wire_map maps condition creg on to more than one
                         # creg in target DAG.
                         raise DAGCircuitError('wire_map maps conditional '
                                               'register onto more than one creg.')
 
-                    if 2**(bit.index) & cond_val:
-                        new_cond_val += 2**(wire_map[bit].index)
+                    if 2**(cond_creg[:].index(bit)) & cond_val:
```

Comment:
Is the `[:]` just there to workaround around a Register not being a list and not having an index method?

Reply:
That's right. (Alternately, we could consider building the bit-> index `dict` like we do elsewhere.)